packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreDuplicateRoutes.spec.ts

```diff
@@ -0,0 +1,154 @@
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const settings = {
+        "Fluid.GarbageCollection.LogUnknownOutboundRoutes": "true",
+        "Fluid.GarbageCollection.WriteDataAtRoot": "true",
+    };
+    const configProvider = mockConfigProvider(settings);
+    settings["Fluid.GarbageCollection.UseDataStoreAliasing"] = "false";
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+    let latestUploadedSummary: ISummaryTree | undefined;
+    let latestAckedSummary: IAckedSummary | undefined;
+
+    /**
+     * Callback that will be called by the document storage service whenever a summary is uploaded by the client.
+     * Update the summary context to include the summary proposal and ack handle as per the latest ack for the
+     * document.
+     */
+    function uploadSummaryCb(summaryTree: ISummaryTree, context: ISummaryContext): ISummaryContext {
+        latestUploadedSummary = summaryTree;
+        const newSummaryContext = { ...context };
+        // If we received an ack for this document, update the summary context with its information. The
+        // server rejects the summary if it doesn't have the proposal and ack handle of the previous
+        // summary.
+        if (latestAckedSummary !== undefined) {
+            newSummaryContext.ackHandle = latestAckedSummary.summaryAck.contents.handle;
+            newSummaryContext.proposalHandle = latestAckedSummary.summaryOp.contents.handle;
+        }
+        return newSummaryContext;
+    }
+
+    const logger = new TelemetryNullLogger();
+
+    async function summarizeOnNewContainerAndGetGCTree(summaryVersion?: string): Promise<IGarbageCollectionState> {
+        await provider.ensureSynchronized();
+        const summarizerClient = await loadSummarizer(
+            provider,
+            runtimeFactory,
+            mainContainer.deltaManager.lastSequenceNumber,
+            summaryVersion,
+            { configProvider },
+        );
+        const summaryResult = await submitAndAckSummary(provider, summarizerClient, logger, false /* fullTree */);
+        latestAckedSummary = summaryResult.ackedSummary;
+        assert(latestUploadedSummary !== undefined, "Did not get a summary");
+        return getGCContentFromTree(latestUploadedSummary);
+    }
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory, { configProvider });
+    };
+
+    const getGCContentFromTree = (summary: ISummaryTree): IGarbageCollectionState => {
```

Comment:
nit: This can be inlined in `summarizeOnNewContainerAndGetGCTree` since there is only one call.